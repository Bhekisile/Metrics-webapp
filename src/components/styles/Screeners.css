.screenersList {
  width: 90%;
  height: 390px;
  margin: 20px auto;
  box-shadow: rgba(99, 99, 99, 0.2) 0 2px 8px 0;
}

.innerList {
  list-style: none;
  display: flex;
  justify-content: space-between;
}

.innerList li {
  height: 55px;
  display: flex;
  align-items: center;
  padding-left: 10px;
  padding-right: 10px;
}

.back-icon {
  padding-top: 20px;
  margin-left: 25px;
  width: 30px;
  height: 30px;
}

@media (min-width: 768px) {
  .screenersList {
    width: 60%;
    height: 390px;
    margin: 20px auto;
    box-shadow: rgba(99, 99, 99, 0.2) 0 2px 8px 0;
  }
}
