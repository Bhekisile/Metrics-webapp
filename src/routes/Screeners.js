import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { HiChevronLeft } from 'react-icons/hi2';
import { Link } from 'react-router-dom';
import { fetchStockScreener } from '../redux/stockScreener/StockScreenersSlice';
import '../components/styles/Screeners.css';

function Screeners() {
  const dispatch = useDispatch();
  const stockScreeners = useSelector((state) => state.stockScreeners.selectedCompany);

  useEffect(() => {
    dispatch(fetchStockScreener());
  }, [dispatch]);

  if (!stockScreeners) {
    return <div>Loading...</div>;
  }

  return (
    <div className="screenersContain">
      <Link className="back-icon" to="/"><HiChevronLeft /></Link>
      <div key={stockScreeners.symbol} className="screenersList">
        <div style={{ backgroundColor: '#c5d9ed' }}>
          <p
            style={{
              paddingTop: '10px',
              marginBottom: '5px',
              paddingLeft: '10px',
              fontFamily: 'stencil Std, fantasy',
            }}
          >
            {stockScreeners.symbol}
          </p>
          <p
            style={{
              paddingBottom: '10px',
              paddingLeft: '10px',
              fontFamily: 'Chalkduster, fantasy',
            }}
          >
            {stockScreeners.companyName}
          </p>
        </div>
        <ul style={{ backgroundColor: '#f0f6fc' }} className="innerList">
          <li>Beta</li>
          <li>{stockScreeners.beta}</li>
        </ul>
        <ul style={{ backgroundColor: '#c5d9ed' }} className="innerList">
          <li>Price</li>
          <li>{stockScreeners.price}</li>
        </ul>
        <ul style={{ backgroundColor: '#f0f6fc' }} className="innerList">
          <li>Last Annual Dividend</li>
          <li>{stockScreeners.lastAnnualDividend}</li>
        </ul>
        <ul style={{ backgroundColor: '#c5d9ed' }} className="innerList">
          <li>Volume</li>
          <li>{stockScreeners.volume}</li>
        </ul>
        <ul style={{ backgroundColor: '#f0f6fc' }} className="innerList">
          <li>Exchange</li>
          <li>{stockScreeners.exchange}</li>
        </ul>
        <ul style={{ backgroundColor: '#c5d9ed' }} className="innerList">
          <li>Market Cap</li>
          <li>{stockScreeners.marketCap}</li>
        </ul>
      </div>
    </div>
  );
}

export default Screeners;
