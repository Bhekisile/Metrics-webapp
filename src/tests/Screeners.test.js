import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Screeners from '../routes/Screeners';

const mockStore = configureStore([]);

test('Screeners component matches snapshot', () => {
  const store = mockStore({
    stockScreeners: { selectedCompany: [] },
  });

  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Screeners />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
