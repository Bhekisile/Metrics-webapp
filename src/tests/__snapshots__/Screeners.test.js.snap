// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screeners component matches snapshot 1`] = `
<div>
  <div
    class="screenersContain"
  >
    <a
      class="back-icon"
      href="/"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.72 12.53a.75.75 0 010-1.06l7.5-7.5a.75.75 0 111.06 1.06L9.31 12l6.97 6.97a.75.75 0 11-1.06 1.06l-7.5-7.5z"
          fill-rule="evenodd"
        />
      </svg>
    </a>
    <div
      class="screenersList"
    >
      <div
        style="background-color: rgb(197, 217, 237);"
      >
        <p
          style="padding-top: 10px; margin-bottom: 5px; padding-left: 10px; font-family: stencil Std, fantasy; height: 40px;"
        />
        <p
          style="padding-bottom: 10px; padding-left: 10px; font-family: Chalkduster, fantasy; height: 40px;"
        />
      </div>
      <ul
        class="innerList"
        style="background-color: rgb(240, 246, 252);"
      >
        <li>
          Beta
        </li>
        <li />
      </ul>
      <ul
        class="innerList"
        style="background-color: rgb(197, 217, 237);"
      >
        <li>
          Price
        </li>
        <li />
      </ul>
      <ul
        class="innerList"
        style="background-color: rgb(240, 246, 252);"
      >
        <li>
          Last Annual Dividend
        </li>
        <li />
      </ul>
      <ul
        class="innerList"
        style="background-color: rgb(197, 217, 237);"
      >
        <li>
          Volume
        </li>
        <li />
      </ul>
      <ul
        class="innerList"
        style="background-color: rgb(240, 246, 252);"
      >
        <li>
          Exchange
        </li>
        <li />
      </ul>
      <ul
        class="innerList"
        style="background-color: rgb(197, 217, 237);"
      >
        <li>
          Market Cap
        </li>
        <li />
      </ul>
    </div>
  </div>
</div>
`;
